third_party/nvidia/hopper/lib/Transforms/WarpSpecialization/TMEMAlloc1D.cpp

```diff
@@ -185,13 +184,81 @@ void generate1DAllocations(OpBuilder &builder, Operation *producer) {
         mlir::cast<mlir::IntegerAttr>(consumer->getAttr("ttg.partition"))
             .getInt();
     if (producerPartition != consumerParition) {
-      TMEM1DAllocator(builder).replaceWith1DTMEM(producer, consumer);
+      TMEM1DAllocator(builder).replaceWith1DTMEM(producer, consumer,
+                                                 allocOpBuffer);
     }
   }
   // Delete tmem.start
   producer->removeAttr("tmem.start");
 }
 
+ttg::MemDescReinterpretOp
+sliceAndReinterpretTMEMBuffer(OpBuilder &builder, ttng::TMEMAllocOp allocOp,
+                              int offset, size_t blockN) {
+  auto allocType = allocOp.getType();
+  auto shape = allocType.getShape();
+  auto oldBlockN = shape[1];
+  // TODO(njriasan): Support 1x128x128?
```

Comment:
This is a very minor fix, but I'm going to do this as a followup to minimize PR size. The example task is `1x128x128`